File: Server/SmartSolarMicrogrid.Api/DTOs/Reservations/CancelReservationRequest.cs
/*
 * SE4040 - Enterprise Application Development
 * Smart Solar Microgrid Trading System
 * File: CancelReservationRequest.cs
 * Purpose: Receive optional client remarks when cancelling an existing energy reservation.
 */

namespace SmartSolarMicrogrid.Api.DTOs.Reservations;

public sealed class CancelReservationRequest
{
    public string? Reason { get; set; }
}

File: Android/app/src/main/java/com/example/smartsolarmicrogridtradingsystem/feature/reservationqr/model/CancelReservationRequest.kt
package com.example.smartsolarmicrogridtradingsystem.feature.reservationqr.model

import org.json.JSONObject

/**
 * Optional remarks payload when cancelling a scheduled energy reservation via POST /api/reservations/{id}/cancel.
 *
 * @property reason Optional cancellation justification provided by the prosumer.
 */
data class CancelReservationRequest(
    val reason: String? = null
) {
    /**
     * Serializes this request into a [JSONObject] matching backend C# property conventions.
     */
    fun toJson(): JSONObject {
        return JSONObject().apply {
            if (!reason.isNullOrBlank()) {
                put("reason", reason)
            }
        }
    }
}
